select_thoth_integration.py

```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env python3
+# workflow-helpers
+# Copyright(C) 2020 Francesco Murdaca
+#
+# This program is free software: you can redistribute it and / or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+"""This script run in a workflow task to select Thoth integration workflow to run."""
+
+import json
+import os
+import logging
+
+from thoth.workflow_helpers.utils import retrieve_adviser_document
+from thoth.workflow_helpers.trigger_finished_webhook import trigger_finished_webhook
+from thoth.common.enums import ThothAdviserIntegrationEnum
+from thoth.workflow_helpers.configuration import Configuration
+
+_LOGGER = logging.getLogger("select_thoth_integration")
+
+
+def trigger_integration_workflow() -> None:
+    """Trigger specific workflow depending on Thoth integration type."""
+    metadata = Configuration._THOTH_ADVISER_METADATA
+    source_type = metadata["source_type"]
+
+    if not metadata:
+        _LOGGER.warning("No adviser metadata provided. No actions performed.")
+        return
```

Comment:
These two pieces of code should be switched around.  Metadata None check should come before trying to pull value from it